website/content/docs/k8s/installation/vault/data-integration/webhook-certs.mdx

```diff
@@ -0,0 +1,261 @@
+1. Configure [Vault as the Service Mesh Certificate Provider on Kubernetes](/docs/k8s/installation/vault/data-integration/connect-ca)
+1. Complete the [Bootstrapping the PKI Engine for Controller and Connect Inject Webhooks](#bootstrapping-the-pki-engine-for-controller-and-connect-inject-webhooks) section.
+
+### Bootstrapping the PKI Engine for Controller and Connect Inject Webhooks
+
+First, we need to bootstrap the Vault cluster by enabling and configuring the PKI Secrets Engine to be able to serve TLS certificates for the controller and connect inject webhooks.
+
+* Mount the PKI Secrets Engine for each:
+
+  ```shell-session
+  $ vault secrets enable -path=controller pki
+  ```
+
+  ```shell-session
+  $ vault secrets enable -path=connect-inject pki
+  ```
+
+* Tune the engine mounts to enable longer TTL:
+
+  ```shell-session
+  $ vault secrets tune -max-lease-ttl=87600h controller
+  ```
+
+  ```shell-session
+  $ vault secrets tune -max-lease-ttl=87600h connect-inject
+  ```
+
+* Generate the root CA for each:
+
+  ```shell-session
+  $ vault write -field=certificate controller/root/generate/internal \
+          common_name="<RELEASE NAME>-controller-webhook" \
+          ttl=87600h
+  ```
+
+  ```shell-session
+  $ vault write -field=certificate connect-inject/root/generate/internal \
+          common_name="<RELEASE NAME>-connect-injector" \
+          ttl=87600h
+  ```
+
+
+## Overview
+In a Consul Helm chart configuration that does not use Vault, webhook-cert-manager normally fulfills the role of ensuring that a valid certificate is updated to the mutatingwebhook configuration of either controller or connect inject to ensure that Kubernetes can communicate with each of these services. 
+
+When Vault is configured as the controller and connect inject Webhook Certificate Provider on Kubernetes: 
+  - webhook-cert-manager is no longer deployed to the cluster.
+  - controller and connect inject each get their webhook certificates from its own Vault PKI mount via the injected Vault Agent.
+  - controller and connect inject each need to be configured with its own Vault Role that has necessary permissions to receive certificates from its respective PKI mount.
+  - controller and connect inject each locally update their mutatingwebhookconfigurations so that Kubernetes can relay events.
+  - Vault will manage certificate rotation and each of these webhooks will seamlessly received rotated certificates.
+
+To use Vault as the controller and connect inject Webhook Certificate Provider, we will need to modify the steps outlined in the [Data Integration](/docs/k8s/installation/vault/data-integration) section:
+
+### One time setup in Vault
+  1. Create a Vault policy that authorizes the desired level of access to the secret.
+
+### Setup per Consul datacenter
+  1. (Added) Create Vault PKI roles for controller and connect inject each that establish the domains that each is allowed to issue certificates for.
```

Comment:
I don't understand why "(added)" is here.

Reply:
Oh, I see. Reading back it's showing what is modified from the steps in the Data Integration section. That was not initially clear to me.

Reply:
It would be clearer to just repeat the steps described in Data Integration with modifications as if we are describing a new process. It's too much to ask users, especially new/novice users, to go back and forth in this way.